pandas/io/pickle.py

```diff
@@ -32,12 +39,21 @@ def read_pickle(path):
     ----------
     path : string
         File path
+    compression : {'infer', 'gzip', 'bz2', 'xz', None}, default 'infer'
 
     Returns
     -------
     unpickled : type of object stored in file
     """
 
+    inferred_compression = _get_inferred_compression(path, compression)
+
+    def openfile():
+        if inferred_compression:
+            return _get_handle(path, 'rb', compression=inferred_compression, is_txt=False)
```

Comment:
I don't understand this code quite well, need help:
https://github.com/pydata/pandas/blob/master/pandas/io/common.py#L314

```
        if compat.PY3 and is_txt:
            from io import TextIOWrapper
            f = TextIOWrapper(f, encoding=encoding)
```